### PR TITLE
Add support of NUCLEO_L433RC_P

### PR DESCRIPTION
**Pull request by ABOSTM**
**Summary**
Add support of NUCLEO_L433RC_P

**Validation**
Test Passed:
* Blink
* Analog read
* Analog write (Pin with PWM and pin with DAC)
* Tone
* I2C scan
* I2C EEPROM (CI-Shield)
* I2C Temperature sensor (CI-Shield)
* USB HID
* USB CDC
* SPI LCD (shield adafruit)
* SPI SD (shield adafruit)
* EEPROM emulation
* IWatchdog
* SoftwareSerial

### CODE REVIEW

File: variants/Generic_F103Zx/PeripheralPins.c [hunk 1]
```diff
@@ -33,9 +33,9 @@ WEAK const PinMap PinMap_ADC[] = {
   // {PA_2,  ADC1,  STM_PIN_DATA_EXT(STM_MODE_ANALOG, GPIO_NOPULL, 0, 2, 0)}, // ADC1_IN2
   {PA_2,  ADC2,  STM_PIN_DATA_EXT(STM_MODE_ANALOG, GPIO_NOPULL, 0, 2, 0)}, // ADC2_IN2
   // {PA_2,  ADC3,  STM_PIN_DATA_EXT(STM_MODE_ANALOG, GPIO_NOPULL, 0, 2, 0)}, // ADC3_IN2
-  {PA_3,  ADC1,  STM_PIN_DATA_EXT(STM_MODE_ANALOG, GPIO_NOPULL, 0, 3, 0)}, // ADC1_IN3
+  // {PA_3,  ADC1,  STM_PIN_DATA_EXT(STM_MODE_ANALOG, GPIO_NOPULL, 0, 3, 0)}, // ADC1_IN3
```
Comment (fpistm) [inline variants/Generic_F103Zx/PeripheralPins.c]: Please revert this change

File: variants/Generic_F103Zx/PeripheralPins.c [hunk 2]
```diff
@@ -33,9 +33,9 @@ WEAK const PinMap PinMap_ADC[] = {
...
   // {PA_3,  ADC2,  STM_PIN_DATA_EXT(STM_MODE_ANALOG, GPIO_NOPULL, 0, 3, 0)}, // ADC2_IN3
-  // {PA_3,  ADC3,  STM_PIN_DATA_EXT(STM_MODE_ANALOG, GPIO_NOPULL, 0, 3, 0)}, // ADC3_IN3
+   {PA_3,  ADC3,  STM_PIN_DATA_EXT(STM_MODE_ANALOG, GPIO_NOPULL, 0, 3, 0)}, // ADC3_IN3
```
Comment (fpistm) [inline variants/Generic_F103Zx/PeripheralPins.c]: Please revert this change

File: variants/NUCLEO_L433RC_P/PeripheralPins.c [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+  {PA_11, TIM1,   STM_PIN_DATA_EXT(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF1_TIM1, 4, 0)}, // TIM1_CH4
+  {PA_15, TIM2,   STM_PIN_DATA_EXT(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF1_TIM2, 1, 0)}, // TIM2_CH1
+  {PB_0,  TIM1,   STM_PIN_DATA_EXT(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF1_TIM1, 2, 1)}, // TIM1_CH2N
+  {PB_1,  TIM1,   STM_PIN_DATA_EXT(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF1_TIM1, 3, 1)}, // TIM1_CH3N
+  {PB_3,  TIM2,   STM_PIN_DATA_EXT(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF1_TIM2, 2, 0)}, // TIM2_CH2
+  {PB_6,  TIM16,  STM_PIN_DATA_EXT(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF14_TIM16, 1, 1)}, // TIM16_CH1N
+  {PB_8,  TIM16,  STM_PIN_DATA_EXT(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF14_TIM16, 1, 0)}, // TIM16_CH1
+  {PB_10, TIM2,   STM_PIN_DATA_EXT(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF1_TIM2, 3, 0)}, // TIM2_CH3
+  {PB_11, TIM2,   STM_PIN_DATA_EXT(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF1_TIM2, 4, 0)}, // TIM2_CH4
+  {PB_13, TIM1,   STM_PIN_DATA_EXT(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF1_TIM1, 1, 1)}, // TIM1_CH1N
```
Comment (fpistm) [inline variants/NUCLEO_L433RC_P/PeripheralPins.c]: To save some space, it should be fine to comment some duplicated pins.
As the first one found will be used.
```suggestion
  //  {PB_13, TIM1,   STM_PIN_DATA_EXT(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF1_TIM1, 1, 1)}, // TIM1_CH1N
```

File: variants/NUCLEO_L433RC_P/PeripheralPins.c [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+  {PB_13, TIM15,  STM_PIN_DATA_EXT(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF14_TIM15, 1, 1)}, // TIM15_CH1N
+  {PB_14, TIM1,   STM_PIN_DATA_EXT(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF1_TIM1, 2, 1)}, // TIM1_CH2N
+  {PB_14, TIM15,  STM_PIN_DATA_EXT(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF14_TIM15, 1, 0)}, // TIM15_CH1
+  {PB_15, TIM1,   STM_PIN_DATA_EXT(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF1_TIM1, 3, 1)}, // TIM1_CH3N
```
Comment (fpistm) [inline variants/NUCLEO_L433RC_P/PeripheralPins.c]:
```suggestion
  //  {PB_15, TIM1,   STM_PIN_DATA_EXT(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF1_TIM1, 3, 1)}, // TIM1_CH3N
```

File: variants/NUCLEO_L433RC_P/PeripheralPins.c [hunk 5]
```diff
@@ -0,0 +1,285 @@
...
+#endif
+
+#ifdef HAL_UART_MODULE_ENABLED
+WEAK const PinMap PinMap_UART_CTS[] = {
+  {PA_0,  USART2,  STM_PIN_DATA(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF7_USART2)},
+  {PA_6,  LPUART1, STM_PIN_DATA(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF8_LPUART1)},
+  {PA_6,  USART3,  STM_PIN_DATA(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF7_USART3)},
+  {PA_11, USART1,  STM_PIN_DATA(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF7_USART1)},
+  {PB_4,  USART1,  STM_PIN_DATA(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF7_USART1)},
+  {PB_13, LPUART1, STM_PIN_DATA(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF8_LPUART1)},
```
Comment (fpistm) [inline variants/NUCLEO_L433RC_P/PeripheralPins.c]:
```suggestion
  //  {PB_13, LPUART1, STM_PIN_DATA(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF8_LPUART1)},
```

File: variants/NUCLEO_L433RC_P/PeripheralPins.c [hunk 6]
```diff
@@ -0,0 +1,285 @@
...
+  {PB_14, USART3,  STM_PIN_DATA(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF7_USART3)},
+  {NC,    NP,    0}
+};
+#endif
+
+#ifdef HAL_UART_MODULE_ENABLED
+WEAK const PinMap PinMap_UART_CTS[] = {
+  {PA_0,  USART2,  STM_PIN_DATA(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF7_USART2)},
+  {PA_6,  LPUART1, STM_PIN_DATA(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF8_LPUART1)},
+  {PA_6,  USART3,  STM_PIN_DATA(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF7_USART3)},
```
Comment (fpistm) [inline variants/NUCLEO_L433RC_P/PeripheralPins.c]:
```suggestion
  //  {PA_6,  USART3,  STM_PIN_DATA(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF7_USART3)},
```

File: variants/NUCLEO_L433RC_P/PeripheralPins.c [hunk 7]
```diff
@@ -0,0 +1,285 @@
...
+  {NC,    NP,    0}
+};
+#endif
+
+#ifdef HAL_UART_MODULE_ENABLED
+WEAK const PinMap PinMap_UART_RTS[] = {
+  {PA_1,  USART2,  STM_PIN_DATA(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF7_USART2)},
+  {PA_12, USART1,  STM_PIN_DATA(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF7_USART1)},
+  {PA_15, USART3,  STM_PIN_DATA(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF7_USART3)},
+  {PB_1,  LPUART1, STM_PIN_DATA(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF8_LPUART1)},
```
Comment (fpistm) [inline variants/NUCLEO_L433RC_P/PeripheralPins.c]:
```suggestion
  //  {PB_1,  LPUART1, STM_PIN_DATA(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF8_LPUART1)},
```

File: variants/NUCLEO_L433RC_P/PeripheralPins.c [hunk 8]
```diff
@@ -0,0 +1,285 @@
...
+};
+#endif
+
+//*** SPI ***
+
+#ifdef HAL_SPI_MODULE_ENABLED
+WEAK const PinMap PinMap_SPI_MOSI[] = {
+  {PA_7,  SPI1, STM_PIN_DATA(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF5_SPI1)},
+  {PA_12, SPI1, STM_PIN_DATA(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF5_SPI1)},
+  {PB_5,  SPI1, STM_PIN_DATA(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF5_SPI1)},
```
Comment (fpistm) [inline variants/NUCLEO_L433RC_P/PeripheralPins.c]:
```suggestion
  //  {PB_5,  SPI1, STM_PIN_DATA(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF5_SPI1)},
```

File: variants/NUCLEO_L433RC_P/PeripheralPins.c [hunk 9]
```diff
@@ -0,0 +1,285 @@
...
+  {PC_12, SPI3, STM_PIN_DATA(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF6_SPI3)},
+  {NC,    NP,    0}
+};
+#endif
+
+#ifdef HAL_SPI_MODULE_ENABLED
+WEAK const PinMap PinMap_SPI_MISO[] = {
+  {PA_6,  SPI1, STM_PIN_DATA(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF5_SPI1)},
+  {PA_11, SPI1, STM_PIN_DATA(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF5_SPI1)},
+  {PB_4,  SPI1, STM_PIN_DATA(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF5_SPI1)},
```
Comment (fpistm) [inline variants/NUCLEO_L433RC_P/PeripheralPins.c]:
```suggestion
  //  {PB_4,  SPI1, STM_PIN_DATA(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF5_SPI1)},
```

File: variants/NUCLEO_L433RC_P/PeripheralPins.c [hunk 10]
```diff
@@ -0,0 +1,285 @@
...
+  {PC_11, SPI3, STM_PIN_DATA(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF6_SPI3)},
+  {NC,    NP,    0}
+};
+#endif
+
+#ifdef HAL_SPI_MODULE_ENABLED
+WEAK const PinMap PinMap_SPI_SCLK[] = {
+  {PA_1,  SPI1, STM_PIN_DATA(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF5_SPI1)},
+  {PA_5,  SPI1, STM_PIN_DATA(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF5_SPI1)},
+  {PB_3,  SPI1, STM_PIN_DATA(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF5_SPI1)},
```
Comment (fpistm) [inline variants/NUCLEO_L433RC_P/PeripheralPins.c]:
```suggestion
  //  {PB_3,  SPI1, STM_PIN_DATA(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF5_SPI1)},
```

File: variants/NUCLEO_L433RC_P/PeripheralPins.c [hunk 11]
```diff
@@ -0,0 +1,285 @@
...
+  {PB_13, SPI2, STM_PIN_DATA(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF5_SPI2)},
+  {PC_10, SPI3, STM_PIN_DATA(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF6_SPI3)},
+  {NC,    NP,    0}
+};
+#endif
+
+#ifdef HAL_SPI_MODULE_ENABLED
+WEAK const PinMap PinMap_SPI_SSEL[] = {
+  {PA_4,  SPI1, STM_PIN_DATA(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF5_SPI1)},
+  {PA_4,  SPI3, STM_PIN_DATA(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF6_SPI3)},
```
Comment (fpistm) [inline variants/NUCLEO_L433RC_P/PeripheralPins.c]:
```suggestion
  //  {PA_4,  SPI3, STM_PIN_DATA(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF6_SPI3)},
```

File: variants/NUCLEO_L433RC_P/PeripheralPins.c [hunk 12]
```diff
@@ -0,0 +1,285 @@
...
+  {PA_15, SPI1, STM_PIN_DATA(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF5_SPI1)},
```
Comment (fpistm) [inline variants/NUCLEO_L433RC_P/PeripheralPins.c]:
```suggestion
  //  {PA_15, SPI1, STM_PIN_DATA(STM_MODE_AF_PP, GPIO_PULLUP, GPIO_AF5_SPI1)},
```

File: variants/Generic_F103Zx/PeripheralPins.c (hunk 1 above)
Comment (ABOSTM) [reply variants/Generic_F103Zx/PeripheralPins.c]: Done

File: variants/Generic_F103Zx/PeripheralPins.c (hunk 2 above)
Comment (ABOSTM) [reply variants/Generic_F103Zx/PeripheralPins.c]: Done

File: variants/NUCLEO_L433RC_P/PeripheralPins.c (hunk 3 above)
Comment (ABOSTM) [reply variants/NUCLEO_L433RC_P/PeripheralPins.c]: Done

File: variants/NUCLEO_L433RC_P/PeripheralPins.c (hunk 4 above)
Comment (ABOSTM) [reply variants/NUCLEO_L433RC_P/PeripheralPins.c]: Done

File: variants/NUCLEO_L433RC_P/PeripheralPins.c (hunk 5 above)
Comment (ABOSTM) [reply variants/NUCLEO_L433RC_P/PeripheralPins.c]: Done

File: variants/NUCLEO_L433RC_P/PeripheralPins.c (hunk 9 above)
Comment (ABOSTM) [reply variants/NUCLEO_L433RC_P/PeripheralPins.c]: Done

File: variants/NUCLEO_L433RC_P/PeripheralPins.c (hunk 10 above)
Comment (ABOSTM) [reply variants/NUCLEO_L433RC_P/PeripheralPins.c]: Done

File: variants/NUCLEO_L433RC_P/PeripheralPins.c (hunk 11 above)
Comment (ABOSTM) [reply variants/NUCLEO_L433RC_P/PeripheralPins.c]: Done

File: variants/NUCLEO_L433RC_P/PeripheralPins.c (hunk 12 above)
Comment (ABOSTM) [reply variants/NUCLEO_L433RC_P/PeripheralPins.c]: Done

File: variants/NUCLEO_L433RC_P/PeripheralPins.c (hunk 8 above)
Comment (ABOSTM) [reply variants/NUCLEO_L433RC_P/PeripheralPins.c]: Done
